feature-settings-impl/src/main/java/io/novafoundation/nova/feature_settings_impl/domain/NetworkManagementInteractor.kt

```diff
@@ -0,0 +1,37 @@
+package io.novafoundation.nova.feature_settings_impl.domain
+
+import io.novafoundation.nova.common.utils.mapList
+import io.novafoundation.nova.runtime.ext.defaultComparatorFrom
+import io.novafoundation.nova.runtime.multiNetwork.ChainRegistry
+import io.novafoundation.nova.runtime.multiNetwork.chain.model.Chain
+import io.novasama.substrate_sdk_android.wsrpc.state.SocketStateMachine
+import kotlinx.coroutines.flow.Flow
+import kotlinx.coroutines.flow.map
+
+class NetworkState(
+    val chain: Chain,
+    val connectionState: SocketStateMachine.State?
+)
+
+interface NetworkManagementInteractor {
+
+    fun defaultNetworksFlow(): Flow<List<NetworkState>>
+}
+
+class RealNetworkManagementInteractor(
+    private val chainRegistry: ChainRegistry
+) : NetworkManagementInteractor {
+
+    override fun defaultNetworksFlow(): Flow<List<NetworkState>> {
+        return chainRegistry.currentChains
+            .mapList {
+                val connection = chainRegistry.getConnectionInstantlyOrNull(it.id)
```

Comment:
This doesnt seem reactive. I think one of the requirements to support reactive updates of the state. So we can do 

```kotlin
chainRegistry.currentChains.flatMapLatest { chains ->
                chains.map {chain ->
                    chainRegistry.getConnection(chain.id).state.map {
                        NetworkState(chain,  it)
                    }
                }.combine()
            }
```

Also as you already spotted `getConnection` ensures the connection is active (by making it active if it is not) so you created `getConnectionInstantlyOrNull`. 
I suggest renaming `getConnection` to `getActiveConnection` and `getConnectionInstantly` to `getConnection` for better transparency of the code